models/users.js

```diff
@@ -79,20 +79,31 @@ const fetchUsers = async (query) => {
 }
 
 /**
- * Fetches the user data from the passes userId
+ * Fetches the user data from the the provided username or userId
  *
- * @param userId { string }: User id
+ * @param { Object }: Object with username and userId, any of the two can be used
  * @return {Promise<{userExists: boolean, user: <userModel>}|{userExists: boolean, user: <userModel>}>}
  */
-const fetchUser = async (userId) => {
+const fetchUser = async ({ userId, username }) => {
   try {
-    const user = await userModel.doc(userId).get()
+    let userData, id
+    if (username) {
+      const user = await userModel.where('username', '==', username).limit(1).get()
 
+      user.forEach(doc => {
+        id = doc.id
+        userData = doc.data()
+      })
+    } else {
```

Comment:
Put a check for userId, both values can also be null.

Reply:
I didn't understand how can both values be null?

Reply:
Edge case